src/main/java/cs2103/v15_1j/jimjim/JJStorage.java

```diff
@@ -0,0 +1,92 @@
+package cs2103.v15_1j.jimjim;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+
+class JJStorage implements Storage {
+	private String fileName;
+	private File saveFile;
+	private List<TaskEvent> list;
+	
+	public JJStorage() throws IOException  {
+		fileName = "data.xml";
+		saveFile = new File(fileName);
+		initializeList();
+	}
+	
+	
+	public JJStorage(String fileName) throws IOException {
+		this.fileName = fileName;
+		saveFile = new File(fileName);
+		initializeList();
+	}
+	
+	private void initializeList() throws IOException {
+		try {
+	        JAXBContext context = JAXBContext.newInstance(TaskEventWrapper.class);
+	        Unmarshaller um = context.createUnmarshaller();
+	        
+	        // Reading XML from the file and unmarshalling.
+	        TaskEventWrapper wrapper = (TaskEventWrapper) um.unmarshal(saveFile);
+	        list.clear();
+	        list.addAll(wrapper.getTaskEvents());
+		} catch (Exception e) {
+			saveFile.createNewFile();
```

Comment:
Ah I see you create the file here. But I still don't like this pattern. In my opinion we should try to avoid Exception as best as we can, so we should check whether saveFile exists and react accordingly instead of guessing its existence based on the throwing of exception (exception may be thrown by other causes)